src/platformdirs/macos.py

```diff
@@ -0,0 +1,69 @@
+import os
+import sys
+
+from .api import PlatformDirsABC
+
+
+class MacOS(PlatformDirsABC):
+    """
+    Platform directories for the macOS operating system. Follows the guidance from `Apple documentation
+    <https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/MacOSXDirectories/MacOSXDirectories.html>`_.
+    Makes use of the :meth:`appname <platformdirs.api.PlatformDirsABC.appname>` and
+    :meth:`version <platformdirs.api.PlatformDirsABC.version>`.
+    """
+
+    @classmethod
+    def is_active(cls) -> bool:
+        """:return: a check to detect if macOS platform is currently active"""
+        return sys.platform == "darwin"
+
+    @property
+    def user_data_dir(self) -> str:
+        """:return: data directory tied to the user, e.g. ``~/Library/Application Support/$appname/$version``"""
+        return self._path_with_app_name_version("~/Library/Application Support/", expand=True)
+
+    def _path_with_app_name_version(self, of: str, *, expand: bool = False) -> str:
+        params = [os.path.expanduser(of) if expand else of]
+        if self.appname:
+            params.append(self.appname)
+            if self.version:
+                params.append(self.version)
+        return os.path.join(of, *params)
```

Comment:
why is of still passed if its part of params?

Reply:
I guess shouldn't but why are the tests not catching this bug 🤔